### ISSUE TITLE
Loading and using a trained model from C++

### ISSUE THREAD

**Issue by yha879** [tflearn/tflearn#661]
I am a newbie to both tensorflow and tflearn.  But I found the tflearn tutorial using the titanic example to be very helpful in getting things started in python and helping me with my research.  It was also pretty straight forward to save a trained model, then loading it again from another python script to make prediction.   

It would be tremendously helpful to have a full example of saving/exporting the titanic model created in python then loading the model and making prediction in C++ code. (C++ code example of preparing Jack Dawson and Rose DeWitt Bukater data to passing them into the model to make prediction of their survival rate)

Thank you in advance!!

**Comment by aymericdamien** [comment]
To load a model and make prediction in C++, you need to "freeze" the graph model, you can find more info here: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/tools/freeze_graph.py

**Comment by yha879** [comment]
Thank you for your response. Is there also C++ side of the code to load the model and predict?  It seems many people would benefit from an end to end example.

Sent from Yahoo Mail on Android 
 
  On Wed, Mar 15, 2017 at 1:16 pm, Aymeric Damien<notifications@github.com> wrote:   
To load a model and make prediction in C++, you need to "freeze" the graph model, you can find more info here: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/tools/freeze_graph.py

**Comment by davideboschetto** [comment]
Do we still need to freeze the model to load it in c++?

**Comment by hjchai** [comment]
@davideboschetto No, you don't. I found [this](https://stackoverflow.com/questions/35508866/tensorflow-different-ways-to-export-and-run-graph-in-c/43639305#43639305) discussion very helpful. You can load checkpoint file to C++ and do inference thereafter.

**Comment by sathyamoorthyrr** [comment]
> @davideboschetto No, you don't. I found [this](https://stackoverflow.com/questions/35508866/tensorflow-different-ways-to-export-and-run-graph-in-c/43639305#43639305) discussion very helpful. You can load checkpoint file to C++ and do inference thereafter.

@davideboschetto 
1. Loading Graph and Checkpoint files in C++ and doing inference
2. Loading frozen graph(contains graph/final checkpoints) and doing inference in C++
Any performance or speed improvements possible with one another?
